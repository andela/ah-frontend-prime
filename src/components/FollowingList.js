import React from "react";

const FollowingList = props => {
  const { following } = props;
  return (
    <div className="modal-dialog" role="document">
      <div className="modal-content">
        <div className="modal-header">
          <h5 className="modal-title">My Following</h5>
          <button
            type="button"
            className="close"
            data-dismiss="modal"
            aria-label="Close"
          >
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div className="modal-body">
          {following && following.length > 0
            ? following.map(item => (
                <div
                  className="row"
                  id="following"
                  key={item.username}
                  onClick={() => {
                    document.location.href = `/profile/${item.username}`;
                  }}
                  key={item.username}
                >
                  <button className="col col-lg-3" style={{ border: "none" }}>
                    <img
                      src={
                        item.image
                          ? `${item.image}`
                          : "https://cdn.pixabay.com/photo/2013/07/13/10/07/man-156584__340.png"
                      }
                      alt="profile"
                      className="profile-page"
                    />
                  </button>
                  <div className="col col-lg-9">
                    <h5>{item.username}</h5>
                    <p>{item.full_name}</p>
                  </div>
                  <br />
                </div>
              ))
            : "No one is on the following List"}
        </div>
        <div class="modal-footer" />
      </div>
    </div>
  );
};

export default FollowingList;
