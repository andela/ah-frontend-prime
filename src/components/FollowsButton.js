import React from "react";

const FollowsButton = props => {
  const { isfollowing, unfollowUser, followUser } = props;
  return (
    <button
      onClick={
        isfollowing
          ? unfollowUser.bind(this, sessionStorage.getItem("userview_name"))
          : followUser.bind(this, sessionStorage.getItem("userview_name"))
      }
      className="btn btn-outline-success"
    >
      {isfollowing ? "Unfollow" : "Follow"}
    </button>
  );
};

export default FollowsButton;
