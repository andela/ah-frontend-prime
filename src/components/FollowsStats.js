import React from "react";
import FollowingList from "./FollowingList";
import FollowersList from "./FollowersList";

const FollowsStats = props => {
  const { following_count, followers_count, followers, following } = props;
  return (
    <div className="row">
      <div
        className="modal fade"
        id="exampleModal"
        tabIndex={-1}
        role="dialog"
        aria-labelledby="exampleModalLabel"
        aria-hidden="true"
      >
        <FollowingList following={following} />
      </div>
      <div
        className="modal fade"
        id="exampleModal1"
        tabIndex={-1}
        role="dialog"
        aria-labelledby="exampleModalLabel"
        aria-hidden="true"
      >
        <FollowersList followers={followers} />
      </div>
      <a
        className="col-md-4"
        data-toggle="modal"
        data-target="#exampleModal1"
        href="#"
      >
        {followers_count === 1 ? "Follower" : "Followers"}
        {" : "}
        <button className="stats">
          <strong>{followers_count ? followers_count : 0} </strong>
        </button>
      </a>
      <a
        className="col-md-4"
        data-toggle="modal"
        data-target="#exampleModal"
        href="#"
      >
        Following{" : "}
        <button className="stats">
          <strong>{following_count ? following_count : "0"} </strong>
        </button>
      </a>
    </div>
  );
};

export default FollowsStats;
