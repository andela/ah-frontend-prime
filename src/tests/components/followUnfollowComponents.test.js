import { shallow, mount } from "enzyme";
import React from "react";
import FollowingList from "../../components/FollowingList";
import FollowersList from "../../components/FollowersList";
import FollowsStats from "../../components/FollowsStats";
import FollowsButton from "../../components/FollowsButton";

describe("following list popup", () => {
  it("should render without crashing", () => {
    const props = {
      following: []
    };
    const wrapper = shallow(<FollowingList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render without crashing", () => {
    const props = {
      following: ["david"]
    };
    const wrapper = shallow(<FollowingList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("followers list popup", () => {
  it("should render without crashing", () => {
    const props = {
      followers: []
    };
    const wrapper = shallow(<FollowersList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render without crashing", () => {
    const props = {
      followers: ["dan"]
    };
    const wrapper = shallow(<FollowersList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("followers and following statistics", () => {
  const props = {
    followers: [],
    following: [],
    followers_count: 0,
    following_count: 0
  };
  it("should render without crashing", () => {
    const wrapper = shallow(<FollowsStats {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("follow/unfollow button", () => {
  const props = {
    isfollowing: false,
    unfollowUser: jest.fn(),
    followUser: jest.fn()
  };
  it("should render without crashing", () => {
    const wrapper = shallow(<FollowsButton {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("follower click", () => {
  it("should click on a follower", () => {
    const props = {
      followers: ["david"]
    };
    const wrapper = mount(<FollowersList {...props} />);
    wrapper.find("div#follower").simulate("click");
    expect(wrapper).toMatchSnapshot();
  });
});

describe("following click", () => {
  it("should click on a user  that is being followed", () => {
    const props = {
      following: ["david"]
    };
    const wrapper = mount(<FollowingList {...props} />);
    wrapper.find("div#following").simulate("click");
    expect(wrapper).toMatchSnapshot();
  });
});
