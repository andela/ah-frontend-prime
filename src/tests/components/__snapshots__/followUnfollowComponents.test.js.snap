// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`follow/unfollow button should render without crashing 1`] = `ShallowWrapper {}`;

exports[`follower click should click on a follower 1`] = `ReactWrapper {}`;

exports[`followers and following statistics should render without crashing 1`] = `ShallowWrapper {}`;

exports[`followers list popup should render without crashing 1`] = `ShallowWrapper {}`;

exports[`followers list popup should render without crashing 2`] = `ShallowWrapper {}`;

exports[`following click should click on a user  that is being followed 1`] = `ReactWrapper {}`;

exports[`following list popup should render without crashing 1`] = `ShallowWrapper {}`;

exports[`following list popup should render without crashing 2`] = `ShallowWrapper {}`;
